k8s/templates/public-splitter.yaml

```diff
@@ -0,0 +1,139 @@
+    worker_rlimit_nofile 8192;
+
+    events {
+      worker_connections  4096;  ## Default: 1024
+    }
+
+    http {
+      proxy_redirect          off;
+      proxy_set_header        Host            $host;
+      proxy_set_header        X-Real-IP       $remote_addr;
+      proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
+      client_max_body_size    32m;
+      client_body_buffer_size 128k;
+      proxy_connect_timeout   90;
+      proxy_send_timeout      90;
+      proxy_read_timeout      90;
+      proxy_buffers           32 4k;
+      resolver                kube-dns.kube-system.svc.cluster.local  valid=5s;
+
+      map $http_cookie $login {
+        default "no";
+        "~*pmo-kli=1" "yes";
+        "~*pmo-access=([^;]+)" "yes";
+      }
+
+      map $login $target {
+        default "dino-park-front-end-service.{{ .Values.namespace }}.svc.cluster.local:80";
+        "yes" "gatekeeper.{{ .Values.namespace }}.svc.cluster.local:80";
+      }
+
+      map $login $api {
+        default "dino-park-fence-service.{{ .Values.namespace }}.svc.cluster.local:80";
+        "yes" "gatekeeper.{{ .Values.namespace }}.svc.cluster.local:80";
+      }
+
+      map $login $pictures {
+        default "dino-park-fossil-service.{{ .Values.namespace }}.svc.cluster.local:80";
+        "yes" "gatekeeper.{{ .Values.namespace }}.svc.cluster.local:80";
+      }
+
+      map $login $github {
+        default "dino-park-whoami-service.{{ .Values.namespace }}.svc.cluster.local:80";
+        "yes" "gatekeeper.{{ .Values.namespace }}.svc.cluster.local:80";
+      }
+
+      server {
+        listen 80;
+
+        location /login {
+          proxy_pass http://gatekeeper.{{ .Values.namespace }}.svc.cluster.local:80/;
+        }
+
+        location /logout {
+          proxy_pass http://gatekeeper.{{ .Values.namespace }}.svc.cluster.local:80/oauth/logout/?redirect=/;
+        }
+
+        location /_/ {
+          proxy_pass http://dino-park-fence-service.{{ .Values.namespace }}.svc.cluster.local:80;
+        }
+
+        location /oauth {
+          proxy_pass http://gatekeeper.{{ .Values.namespace }}.svc.cluster.local:80;
+        }
+
+        location /api/v4/ {
+          proxy_pass http://$api;
+        }
+
+        location  /whoami/github/username/ {
+          proxy_pass http://$github;
+        }
+
+        location /avatar/get/ {
+          proxy_pass http://$pictures;
+        }
+
+        location / {
+          proxy_pass http://$target;
+        }
+      }
+    }
+
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: public-splitter
+  namespace: {{ .Values.namespace }}
+  labels:
+    app: public-splitter
+spec:
+  replicas: {{ .Values.router_replicas }}
+  template:
+    metadata:
+      labels:
+        app: public-splitter
+    spec:
+      volumes:
+      - name: public-splitter
+        configMap:
+          name: public-splitter
+      containers:
+      - name: public-splitter
+        image: nginx
```

Comment:
no pinning this to a specific label, will make it use the newest image available, which at certain point can auto-break the application. I recommend to use a tag here

Reply:
:+1: